### DISCUSSION TITLE
Collapsible page section names not returned in search results?

### DISCUSSION THREAD

**Discussion by AlanTheta** [microsoft-search/pnp-modern-search#3544] [Q&A]
Hi there,
We have collapsible SharePoint page sections however the section names aren't returned in PnP modern search results, although the text body under the section is returned. Is there a way to customise the query or refinement filters to include the section names?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/111036696/e74c1c99-61da-427d-9a7d-3848f0f8df1f)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/111036696/67af7368-5011-46e9-8c8c-45efd0ca9115)
Cheers, Alan

**Comment by wobba** [comment]
Can you share a bit more of your setup and what you are trying to do? How is your data set up for mapping, column->crawled property->refinable string managed property?

**Reply by AlanTheta** [reply]
For now its basic OOTB so we haven't yet added any custom slots / refinable string managed properties. I can see there is a system-defined "SectionNames" managed property however this isn't searchable or refinable. 

![Screenshot 2024-02-22 150617](https://github.com/microsoft-search/pnp-modern-search/assets/111036696/936505b2-fd2e-47d5-b92d-31847270f529)

It's for FAQ-style pages where you want the user to be able to search for terms that will be in the section names/titles and it return a result for the page (or potentially, the section itself if possible) in the PnP Search Results. At the moment it will only return a result if we duplicate the section name itself into its Text web part.

**Reply by wobba** [reply]
If you want to bring something back into the UX, you need to configure the property to be returned, and then edit a slot/template to make it show. If you want section names as a filter you need to create a crawled property to managed property mapping with correct ones.  I suggest starting to read https://learn.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search.

**Reply by kasperbolarsen** [reply]
I tried to add a few sections to a page and as far as I can tell the Section Names are returned in the SPFxExtensionJson and I doubt you will be able to use that for filtering.
Did you manage to get the Section Names to become a crawled property?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/df71b257-8c71-46c7-a621-9f7afb7e3496)

**Reply by AlanTheta** [reply]
Unfortunately not - OOTB the SectionNames managed property is mapped to the ChapterTitle crawled property. I went through the link (thanks Wobba) and have tried creating a custom managed property and mapping to this crawled property and likewise creating a crawled property to custom managed property mapping, but to no avail. 

I'm thinking this is just a frustrating limitation as the text within the Section Names simply isn't captured in anything that can be mapped/crawled

**Reply by wobba** [reply]
I reread your question. The OOB sectionnames property is/was used for headers in Word documents and is used as part of Stream video chapters. Not for sections in a page. If the section names are not returned as free text, this is about the crawler in SharePoint. And doubt there is a way to pull it out as a separate property.
